File: Programação/Back-End PHP/05/mundocao/06 - Classe_interface/Cachorro.php
<?php

    class Cachorro extends Animal{
        public string $nome;
                
        public function darEspecie(string $especie){
            $this->especie = $especie;
        }

        public function retornarEspecie(){
            return $this->especie;
        }

        public function Som()
        {
            echo "Au Au!" . PHP_EOL;
        }

    }

?>

File: 02-programacao/java/02-programacao-orientada-objetos/05-interfaces/exemplo/iaula-interface/src/animal/Cachorro.java
package animal;

import contrato.Cavador;

public class Cachorro extends Animal implements Cavador {
    // Construtor
    public Cachorro(String nome, String especie) {
        super(nome, especie);
    }

    @Override
    public void cavar() {
        System.out.println("O Cachorro está cavando.");
    }

    @Override
    public void emitirSom() {
        System.out.println("O Cachorro está latindo.");
    }
}
